src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+  g_autofree gchar *summary_url = NULL;
+
+  if (url_override != NULL)
+    url = g_strdup (url_override);
+  else if (!ostree_repo_remote_get_url (repo, remote_name, &url, error))
+    return FALSE;
+
+  summary_url = g_strconcat (url, "/summary", NULL);
+  return commit_checksum_from_any_summary (repo,
+                                           remote_name,
+                                           ref,
+                                           summary_url,
+                                           cancellable,
+                                           out_checksum,
+                                           out_extensions,
+                                           error);
+}
+
+static gboolean
+fetch_commit_checksum (OstreeRepo *repo,
+                       GCancellable *cancellable,
+                       const gchar *remote_name,
+                       const gchar *ref,
+                       const gchar *url_override,
+                       gchar **out_checksum,
+                       EosExtensions **out_extensions,
+                       GError **error)
+{
+  if (commit_checksum_from_extensions_ref (repo,
+                                           cancellable,
+                                           remote_name,
+                                           ref,
+                                           url_override,
+                                           out_checksum,
+                                           out_extensions,
+                                           error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_extensions_summary (repo,
+                                               cancellable,
+                                               remote_name,
+                                               ref,
+                                               url_override,
+                                               out_checksum,
+                                               out_extensions,
+                                               error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_summary (repo,
+                                    cancellable,
+                                    remote_name,
+                                    ref,
+                                    url_override,
+                                    out_checksum,
+                                    out_extensions,
+                                    error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (url_override != NULL)
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s with URL %s",
+                 ref, remote_name, url_override);
+  else
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s",
+                 ref, remote_name);
```

Comment:
I wonder if it would be worth buffering up the error messages from each of the attempts to get the checksum. For debugging purposes it would be useful to know which ones failed and why, especially if I was expecting one of them to succeed.